lib/ng-openapi-gen.ts

```diff
@@ -83,24 +83,32 @@ export class NgOpenApiGen {
         }
       }
 
-      // Generate each service and function
+      // Generate each service
       const generateServices = this.options.services ?? true;
       const services = [...this.services.values()];
       for (const service of services) {
         if (generateServices) {
           this.write('service', service, service.fileName, 'services');
         }
-        for (const op of service.operations) {
-          for (const variant of op.variants) {
-            this.write('fn', variant, variant.importFile, variant.importPath);
-          }
-        }
+      }
+
+      // Generate each function
+      const functions = services.reduce((acc, service) => [
+        ...acc,
+        ...service.operations.reduce((opAcc, operation) => [
+          ...opAcc,
+          ...operation.variants
+        ], [])
+      ], []);
```

Comment:
Here I wanted to use `.flatMap` but since we're targeting `es6` that wasn't available, so I used some `.reduce`s.

Looks a bit more complicated, but gets the job done 😃